core/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTest.java

```diff
@@ -0,0 +1,353 @@
+        }
+
+        if (queryBuilder.fields().size() > 1 && (! "".equals(queryBuilder.text()))) {
+            assertTrue("Query should have been BooleanQuery but was " + query.getClass().getName(), query instanceof BooleanQuery);
+
+            BooleanQuery boolQuery = (BooleanQuery) query;
+            if (queryBuilder.lowercaseExpandedTerms()) {
+                for (BooleanClause clause : boolQuery.clauses()) {
+                    if (clause.getQuery() instanceof TermQuery) {
+                        TermQuery inner = (TermQuery) clause.getQuery();
+                        assertThat(inner.getTerm().bytes().toString(), is(inner.getTerm().bytes().toString().toLowerCase()));
+                    }
+                }
+            }
+
+            if (queryBuilder.minimumShouldMatch() != null) {
+                if ("1".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("-1".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("75%".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("-25%".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("2<75%".equals(queryBuilder.minimumShouldMatch())) {
+                    if (shouldClauses(boolQuery) > 2) {
+                        assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                    }
+                } else if ("2<-25%".equals(queryBuilder.minimumShouldMatch())) {
+                    if (shouldClauses(boolQuery) > 2) {
+                        assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                    }
+                }
+            }
+        } else if (queryBuilder.fields().size() == 1 && (! "".equals(queryBuilder.text()))) {
+            assertTrue("Query should have been TermQuery but was " + query.getClass().getName(), query instanceof TermQuery);
+            
+            TermQuery termQuery = (TermQuery) query;
+            assertThat(termQuery.getTerm().field(), is(queryBuilder.fields().keySet().iterator().next()));
+
+            if (queryBuilder.lowercaseExpandedTerms()) {
+                assertThat(termQuery.getTerm().bytes().toString(), is(termQuery.getTerm().bytes().toString().toLowerCase()));
+            }
+
+        } else if ("".equals(queryBuilder.text())) {
+            assertTrue("Query should have been MatchNoDocsQuery but was " + query.getClass().getName(), query instanceof MatchNoDocsQuery);
+        } else {
+            fail("Encountered lucene query type we do not have a validation implementation for in our SimpleQueryStringBuilderTest");
+        }
```

Comment:
I wonder if these checks for the lucene query are necessary given that we already have createExpectedQuery and the comparison between that query and the result of toQuery